picoquic/quicctx.c

```diff
@@ -2806,6 +2807,11 @@ picoquic_cnx_t* picoquic_create_cnx(picoquic_quic_t* quic,
             cnx->local_parameters.enable_loss_bit = quic->default_lossbit_policy;
             cnx->local_parameters.enable_multipath = quic->default_multipath_option & 1;
             cnx->local_parameters.enable_simple_multipath = (quic->default_multipath_option >> 1) & 1;
+            /* Apply the defined MTU MAX instead of default, if specified */
+            if (cnx->quic->mtu_max > 0)
+            {
+                cnx->local_parameters.max_packet_size = cnx->quic->mtu_max;
+            }
```

Comment:
isn't this always incorrect, since the MTU max to be probed includes the UDP + IPv4/IPv6 header sizes and the max_packet_size is just the maximum quic packet size? and when set per connection, we already know the address of the path thus whether it's IPv4 or IPv6 and can derive the accurate maximum quic packet size accepted?

Reply:
The "MTU MAX" in the code is always "UDP max payload size", consistent with the definitions in QUIC transport. So maybe we should set the name to "udp_payload_max".

Yes, there is an issue of whether this should be optimized for V4 or V6. Here we have a one-size-fits-all, which is a limitation. We could have two parameters per QUIC context, one for V6 and one for V4, and pick the one for the address family of the path's destination address.

Reply:
oh okay! i misunderstood that until now. guess that means there's a problem with the name for sure.

given that `max_udp_payload_size`, here `max_packet_size`, is a transport parameter and thus must be provided by both applications. i think the cleanest way to do this is to do away with `mtu_max` and expect the opposing client provided the maximum ipv4-derived (because that truly would be the maximum) quic packet size, and then per path once we know if it's ipv4 or 6, we derive the correct maximum IP + UDP + QUIC MTU to probe for.

i think having these 2 opposing inputs is confusing and unnecessary, and also requires defining their interplay. but if necessary for backwards compatibility in the api, if both are set, we clamp to the minimum. 

when `mtu_max` is larger than the opposing party's `max_packet_size`, there's no sense in probing for an MTU higher because the opposing party already said it can't accept packets any larger. but if `mtu_max` is less than the opposing `max_packet_size`, maybe there's some sensible reason the application wants to underutilize the available packet size.

Reply:
I am thinking of the people using the API. We make the API more complex, as in:
```
void picoquic_set_mtu_max(picoquic_quic_t* quic, uint32_t mtu_max_ipv4, uint32_t mtu_max_ipv6);
```
I can see people puzzling their head about that. Another possibility is to slightly change the semantics of `picoquic_set_mtu_max`, so it sets the `max packet size`, including the IP and UDP headers, with the implication that `picoquic_set_mtu_max(quic, mtu_max)` is equivalent to:
```
payload_max_ipv4 = mtu_max - 20 - 8;
payload_max_ipv6 = mtu_max - 40 - 8;
```
So the default would be `picoquic_set_mtu_max(quic, 1500)`, or for 5G `picoquic_set_mtu_max(quic, 1420)`.

Would that work?

Reply:
The PR #1231 revises the implementation of `mtu_max` to be interpreted as the conventional definition of the MTU.

I think it is important to have a parameter that may override the values proposed by the peer. In the general case, the two peers are independent, not the same admin. It makes sense for the server to limit the packet size that the peer can send, because the server will have to deal with the incoming packets.

Reply:
i think that's the perfect solution